common/persistence/nosql/nosql_execution_store.go

```diff
@@ -90,10 +90,10 @@ func (d *nosqlExecutionStore) CreateWorkflowExecution(
 		return nil, err
 	}
 
-	transferTasks, crossClusterTasks, replicationTasks, timerTasks, err := d.prepareNoSQLTasksForWorkflowTxn(
+	tasksByCategory, err := d.prepareNoSQLTasksForWorkflowTxn(
 		domainID, workflowID, runID,
-		newWorkflow.TransferTasks, newWorkflow.CrossClusterTasks, newWorkflow.ReplicationTasks, newWorkflow.TimerTasks,
-		nil, nil, nil, nil,
+		newWorkflow.TasksByCategory,
+		map[persistence.HistoryTaskCategory][]*nosqlplugin.HistoryMigrationTask{},
```

Comment:
`HistoryMigrationTask` is misleading because this type represents all kind of tasks. Did you name it Migration task because we are migrating to a new schema? Post-migration this won't make sense to readers. HistoryTaskV2 would be a more appropriate name I think

Reply:
Yeah, I'm thinking we should migrate the history task data from `transfer`, `timer` and `replication` columns to `task` and `task_encoding` columns. And after migration, we can delete those 3 columns and get rid of this type. But `HistoryTaskV2` is also a good name.